### PR TITLE
Build: disable link-check for existing medium blog posts

### LINKED ISSUE #10038: Ci: Throttling causes flakyness in link checker
### Apache Iceberg version

1.5.0 (latest release)

### Query engine

None

### Please describe the bug 🐞

When you open a new Iceberg PR, the link checker CI step sometimes errors out because of HTTP 429 (Slow down) errors. This was seen in a [recent PR](https://github.com/apache/iceberg/pull/9963), so opening this issue to track.

### PR DESCRIPTION
**Pull request by manuzhang**
This fixes #10038

### CODE REVIEW

**Review by CsengerG — APPROVED** [review]
Looks good to me

**Comment by nastra** [comment]
While it's great to have retries, it seems that this doesn't solve the issue and the Link checker runs for 48m+. So maybe we should just do `"aliveStatusCodes": [429, 200]`?

**Comment by CsengerG** [comment]
> So maybe we should just do "aliveStatusCodes": [429, 200]?

Is the order of checking the links deterministic? If it is, then that can hide issues (page is not actually present, but by the time you get there it is a 429 --- what are the chances/do we care?). If the order of traversal is not deterministic, then we can perhaps ignore throttling (as you would expect to catch the dead links at some point anyway).

File: site/link-checker-config.json [hunk 1]
```diff
@@ -19,5 +19,8 @@
       "pattern": "^../../",
       "replacement": "{{BASEURL}}/site/docs/"
     }
-  ]
-}
\ No newline at end of file
+  ],
+  "retryOn429": true,
+  "retryCount": 5,
+  "fallbackRetryDelay": "120s"
```
Comment (nastra) [inline site/link-checker-config.json:25]: can we reduce the retries? I think with that retry the last CI run took 48 mins, which is too long IMO

**Comment by manuzhang** [comment]
I'm just disabling link check for existing medium blog posts, which have been causing failure due to 429.

File: site/link-checker-config.json (hunk 1 above)
Comment (manuzhang) [reply site/link-checker-config.json:25]: I don't think it will work anyway. I'm not exploring this approach now.

**Comment by nastra** [comment]
> I'm just disabling link check for existing medium blog posts, which have been causing failure due to 429. WDYT?

Yeah I think this is probably the best for now. Can you remove the retry config and we'll go with the current approach?

**Comment by manuzhang** [comment]
> Can you remove the retry config and we'll go with the current approach?

Done.

**Review by nastra — APPROVED** [review]

**Comment by nastra** [comment]
@manuzhang it looks like there are a few more medium links that haven't been covered. Can you please follow-up on those as well?

**Comment by manuzhang** [comment]
Sure, do we want to run link check on main branch or just PRs for changed docs?

**Comment by nastra** [comment]
@manuzhang can you open up a PR to fix the remaining links please? Also IMO we should be running this for PRs and on main

**Comment by manuzhang** [comment]
@nastra just created #10057

**Pull request merged**